src/content/docs/Standard Library/index.mdx

```diff
@@ -334,31 +334,116 @@ moving the output pointer 1 or 2 steps.
 ## std::io
 
 ### String! readline(stream = io::stdin(), Allocator allocator = allocator::heap())
-Read from a stream (default is stdin) to the next "\n" or to the end of the stream, whatever comes first.
+Read a `String!` from a file stream. Defaults to standard input (stdin). Reads to the next newline or to the end of stream. `Readline` returns an optional, and must be handled accordingly.
 
+```c
+import std::io;
+
+fn void! main() {
+    String! name; 
+
+    name = io::readline();
+    if (catch excuse = name) {
+        return excuse?;
+    }
+
+    io::printfn("The Matrix has you... Follow the white rabbit. Knock, knock, %s.", name);
+}
+```
+
+Note:
 `\r` will be filtered from the String.
 
 ### String! treadline(stream = io::stdin())
-Same as `readline` but uses the temporary allocator.
+Read a `String!` from a file stream. Defaults to standard input (stdin). Reads to the next newline or to the end of stream. `Treadline` returns an optional, and must be handled accordingly. Additionally, the difference from the regular `readline`, is that this uses the temporary allocator.
 
-### void print(x), void printn(x = "")
-Print any value to stdout.
+```c
+import std::io;
 
+fn void! main() {
+    String! name; 
+
+    name = io::treadline();
+    if (catch excuse = name) {
+        return excuse?;
+    }
+
+    io::printfn("Hello %s! Hope you have a great day", name);
+}
```

Comment:
```suggestion
```c3
import std::io;

fn void! main() 
{
    String! name = io::treadline();
    if (catch excuse = name) 
    {
        return excuse?;
    }

    io::printfn("Hello %s! Hope you have a great day", name);
}
```